etc/inc/dyndns.class

```diff
@@ -695,8 +695,14 @@
 						log_error("GratisDNS.dk ({$this->_dnsHost}): DNS update() starting.");
 					}
 					$server = "https://ssl.gratisdns.dk/ddns.phtml";
-					list($hostname, $domain) = explode(".", $this->_dnsHost, 2);
-					curl_setopt($ch, CURLOPT_URL, $server . '?u=' . $this->_dnsUser . '&p=' . $this->_dnsPass . '&h=' . $this->_dnsHost . '&d=' . $domain . '&i=' . $this->_dnsIP);
+					$host = trim($this->_dnsHost);
+					$hostnames = explode(".", $host);
+					$hostnames_count = count($hostnames);
```

Comment:
$hostnames_count is going to be > 1 for "bob.dk" and for "www.bob.dk", this check looks wrong to me.

What does gratisdns API expect to receive on "d" and "h" parameters for both cases? I imagine for "www.bob.dk" it's supposed to be "h=www" and d="bob.dk", right? What about when it's just "bob.dk"? Is h supposed to be blank?